bpf/kmesh/workload/include/backend.h

```diff
@@ -57,24 +57,58 @@ static inline int backend_manager(ctx_buff_t *ctx, backend_value *backend_v)
         return -ENOEXEC;
     }
 
+    return 0;
+}
+
+// through service to access the backend
+static inline int
+service_backend_manager(ctx_buff_t *ctx, backend_value *backend_v, __u32 service_id, service_value *service_v)
+{
+    int ret;
+    address_t target_addr;
+    __u32 user_port = ctx->user_port;
+
+    ret = waypoint_manager(ctx, backend_v->waypoint_addr, backend_v->waypoint_port);
+    if (ret != 0) {
+        BPF_LOG(ERR, BACKEND, "waypoint_manager failed, ret:%d\n", ret);
+        return ret;
+    }
+
 #pragma unroll
-    for (unsigned int i = 0; i < backend_v->port_count; i++) {
-        if (i >= MAX_PORT_COUNT) {
+    for (__u32 i = 0; i < backend_v->service_count; i++) {
+        if (i >= MAX_SERVICE_COUNT) {
             BPF_LOG(WARN, BACKEND, "exceed the max port count\n");
             return -EINVAL;
         }
-
-        if (ctx->user_port == backend_v->service_port[i]) {
-            target_addr.ipv4 = backend_v->ipv4;
-            target_addr.port = backend_v->target_port[i];
-            SET_CTX_ADDRESS(ctx, target_addr);
-            BPF_LOG(DEBUG, BACKEND, "get the backend addr=[%u:%u]\n", target_addr.ipv4, target_addr.port);
-            return 0;
+        if (service_id == backend_v->service[i]) {
+            BPF_LOG(DEBUG, BACKEND, "access the backend by service:%d \n", service_id);
+#pragma unroll
+            for (__u32 j = 0; j < MAX_PORT_COUNT; j++) {
+                if (service_v->service_port[j] != 0 && user_port == service_v->service_port[j]) {
```

Comment:
service_v->service_port[j] != 0 -- no need to check

Reply:
fixed